frontend/src/components/ColorLegend.svelte

```diff
@@ -1,28 +1,42 @@
 <script lang="ts">
-    export let colors;
+    export let styles;
+    import G6, { Graph } from "@antv/g6";
+    import { onMount } from "svelte";
+    import G6Component from "svelte-g6";
+    const options = {
+        container: "mountNode",
+        width: 280,
+        height: 210,
+        workerEnabled: false,
+    };
+
+    let data = {};
+    let counter = 0;
+    onMount(() => {
+        let array = [];
+
+        styles.forEach((event) => {
```

Comment:
Prefer functional programming, i.e.
```
styles.map(event => { id: event[0], size: 10, etc, etc })
```
Have you considered making this a computed value?
```
$: data = styles.map(...)
```

Reply:
> Prefer functional programming, i.e.
> 
> ```
> styles.map(event => { id: event[0], size: 10, etc, etc })
> ```
> 
> Have you considered making this a computed value?
> 
> ```
> $: data = styles.map(...)
> ```

I have changed the code to functional programming.